components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -1199,6 +1204,7 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
     async createPortalSession(ctx: TraceContext): Promise<{}> {
         const user = this.checkUser("createPortalSession");
         const logContext = { userId: user.id };
+        await this.ensureChargebeeApiIsAllowed({ user });
```

Comment:
Playing devil's advocate for a bit: What is the value of purposefully making API methods fail when users are not in the "right mode"?

To me, it's more important the make the UI look exactly correct (e.g. show the relevant UI pages / controls / menu items for whichever mode you're in), but it's less important that every API call is completely fenced-off all the time.

For example:
- What if someone is already using Usage-Based, but they still have some Chargebee subscription? (I know we don't expect this to happen, but if it does, is there value in forbidding the "cancel" API call? Or even forbidding users from opening the Chargebee portal?)

Reply:
To clarify, I'm still happy to go ahead with this change, but I wonder if maybe it would make more sense to always allow certain API calls regardless of your billing mode (e.g. showing the Chargebee portal, and maybe cancelling, while the checkout methods are good to fence off to prevent buying new Chargebee plans once you're on the other side).

Reply:
> but it's less important that every API call is completely fenced-off all the time.

Never trust the client, basically. And as we want to be strict with enforcing that users cannot (accidentally) move back to Chargebee, this just make sense to me.

> is there value in forbidding the "cancel" API call? Or even forbidding users from opening the Chargebee portal?

Ah, those I planned to leave out in this PR. If I haven't, I will do another round to remove those. (Portal is debateable, because people can upgrade there, too!)

Reply:
cmp. [comment](https://github.com/gitpod-io/gitpod/pull/11814#issuecomment-1207714530).